tests/Feature/Router/ErrorHandlingTest.php

```diff
@@ -199,4 +201,35 @@ public function __invoke(): string
             ],
         ], $this->headers());
     }
+
+    /**
+     * @dataProvider nonStringProvider
+     *
+     * @param mixed $given
+     * @param mixed $type
+     */
+    public function test_throws_exception_if_response_is_not_a_string_or_stringable($given, $type): void
+    {
+        $_SERVER['REQUEST_URI'] = 'https://example.org/expected/uri';
+        $_SERVER['REQUEST_METHOD'] = Request::METHOD_GET;
+
+        Router::configure(static function (Handlers $handlers, Routes $routes) use ($given): void {
+            $routes->get('expected/uri', static fn () => $given);
+
+            $handlers->handle(
+                UnsupportedResponseTypeException::class,
+                static fn (UnsupportedResponseTypeException $exception): string => $exception->getMessage(),
+            );
+        });
```

Comment:
Is not the `UnsupportedResponseTypeException` thrown by default?
Is it necessary to define the handler for this test?

Reply:
I wanted to make sure the correct exception was being thrown, otherwise it could be any other exception since we have a built-in handler that catches all unhandled exceptions by default, then returns a generic 500 error and this could end up being a false positive.

The other option was to directly test the class that throws that exception.